src/osdc/Objecter.cc

```diff
@@ -2884,10 +2884,12 @@ int Objecter::_calc_target(op_target_t *t, Connection *con, bool any_change)
     force_resend = true;
   }
 
-  bool unpaused = false;
-  if (t->paused && !target_should_be_paused(t)) {
+  bool unpaused = !target_should_be_paused(t);
+  if (t->paused && unpaused) {
     t->paused = false;
-    unpaused = true;
+  } else {
+    t->paused = !unpaused;
+    unpaused = false;
```

Comment:
I don't really like the use of ```unpaused``` as a temporary variable here, which might get assigned twice because the first assignment was wrong. Can you make the logic a little more explicit? :)